repository/src/test/java/org/alfresco/schedule/AbstractScheduledLockedJobTest.java

```diff
@@ -0,0 +1,208 @@
+public class AbstractScheduledLockedJobTest extends BaseSpringTest
+{
+    private static final int TOTAL_NODES = 9;
+    private static final int NUM_THREADS = 2;
+    private static final long JOB_EXECUTER_LOCK_TTL = 30000L;
+    private static final String ARCHIVE_STORE_URL = "archive://SpacesStore";
+
+    private NodeService nodeService;
+    private TransactionService transactionService;
+    private Repository repository;
+
+    private SchedulerAccessorBean testCleanerAccessor;
+    private JobDetail testCleanerJobDetail;
+
+    private static final Log LOGGER = LogFactory.getLog(AbstractScheduledLockedJobTest.class);
+
+    /**
+     * Sets services and job beans
+     */
+    @Before
+    public void setUp()
+    {
+        nodeService = (NodeService) applicationContext.getBean("nodeService");
+        transactionService = (TransactionService) applicationContext.getBean("transactionComponent");
+        repository = (Repository) applicationContext.getBean("repositoryHelper");
+    }
+
+    @Test
+    public void test() throws SchedulerException, InterruptedException
+    {
+        createAndDeleteNodes(TOTAL_NODES);
+
+        assertTrue("Expected nodes haven't been created", getNumberOfNodesInTrashcan() >= TOTAL_NODES);
+
+        CleanerThread[] threads = new CleanerThread[NUM_THREADS];
+
+        for (int i = 0; i < NUM_THREADS; i++)
+        {
+            CleanerThread t = new CleanerThread(i);
+            threads[i] = t;
+            t.start();
+            Thread.sleep(JOB_EXECUTER_LOCK_TTL);
+        }
+
+        for (Thread t : threads)
+        {
+            t.join();
+        }
+
+        while (getNumberOfNodesInTrashcan() > 0)
+        {
+            Thread.sleep(2000);
+        }
+
+        for (int i = 0; i < threads.length; i++)
+        {
+            CleanerThread t = threads[i];
+            if (t.hasErrors())
+            {
+                fail("An error has occurred when executing multiple cleaner jobs at the same time");
+            }
+        }
+    }
+
+    /**
+     * Creates and deletes the specified number of nodes.
+     *
+     * @param archivedNodes
+     *            Number of nodes to be created and added to trashcan
+     */
+    private void createAndDeleteNodes(int archivedNodes)
+    {
+        AuthenticationUtil.runAsSystem(() -> {
+            RetryingTransactionHelper.RetryingTransactionCallback<Void> txnWork = () -> {
+                for (int i = 0; i < archivedNodes; i++)
+                {
+                    addNodeToTrashcan();
+                }
+                return null;
+            };
+            return transactionService.getRetryingTransactionHelper().doInTransaction(txnWork);
+        });
+    }
+
+    /**
+     * Creates and deletes nodes
+     */
+    private void addNodeToTrashcan()
+    {
+        NodeRef companyHome = repository.getCompanyHome();
+        String name = "Sample (" + System.currentTimeMillis() + ")";
```

Comment:
Does it take more than a millisecond to create a node? It seems like it might be safer to use a UID.

Reply:
Changed from time to UUID.